backend/internal/system/export/service.go

```diff
@@ -0,0 +1,285 @@
+	[]ExportFile, []ExportError) {
+	logger := log.GetLogger().With(log.String("component", "ExportService"))
+	exportFiles := make([]ExportFile, 0, len(applicationIDs))
+	exportErrors := make([]ExportError, 0, len(applicationIDs))
+
+	applicationIDList := make([]string, 0)
+	if len(applicationIDs) == 1 && applicationIDs[0] == "*" {
+		// TODO: Implement GetAllApplications method or pagination
+		apps, err := es.applicationService.GetApplicationList()
+		if err != nil {
+			logger.Warn("Failed to get all applications", log.Any("error", err))
+			return nil, nil
+		}
+		for _, app := range apps.Applications {
+			applicationIDList = append(applicationIDList, app.ID)
+		}
+	} else {
+		applicationIDList = applicationIDs
+	}
+
+	for _, appID := range applicationIDList {
+		// Get the application
+		app, svcErr := es.applicationService.GetApplication(appID)
+		if svcErr != nil {
+			logger.Warn("Failed to get application for export",
+				log.String("appID", appID), log.String("error", svcErr.Error))
+			exportErrors = append(exportErrors, ExportError{
+				ResourceType: "application",
+				ResourceID:   appID,
+				Error:        svcErr.Error,
+				Code:         svcErr.Code,
+			})
+			continue // Skip applications that can't be found
+		}
+
+		// Convert to export format based on options
+		var content string
+		var fileName string
+
+		if options.Format == formatJSON {
+			// Convert to JSON format (could be implemented later)
+			logger.Warn("JSON format not yet implemented, falling back to YAML")
+			options.Format = formatYAML
+		}
+
+		templateContent, err := generateTemplateFromStructV2(app, app.Name)
+		if err != nil {
+			logger.Warn("Failed to generate template from struct",
+				log.String("appID", appID), log.String("error", err.Error()))
+			exportErrors = append(exportErrors, ExportError{
+				ResourceType: "application",
+				ResourceID:   appID,
+				Error:        err.Error(),
+				Code:         "TemplateGenerationError",
+			})
+			continue
+		}
+		content = templateContent
+
+		// Determine file name and folder path based on options
+		fileName = es.generateFileName(app.Name, "application", appID, options)
+		folderPath := es.generateFolderPath("application", options)
+
+		// Create export file
+		exportFile := ExportFile{
+			FileName:     fileName,
+			Content:      content,
+			FolderPath:   folderPath,
+			ResourceType: "application",
+			ResourceID:   appID,
+		}
+		exportFiles = append(exportFiles, exportFile)
+	}
+
+	return exportFiles, exportErrors
+}
+
+func generateTemplateFromStructV2(data interface{}, appName string) (string, error) {
+	rules := templatingRules{
+		Application: &resourceRules{
+			Variables: []string{
+				"InboundAuthConfig[].OAuthAppConfig.ClientID",
+				"InboundAuthConfig[].OAuthAppConfig.ClientSecret",
+			},
+			ArrayVariables: []string{
+				"InboundAuthConfig[].OAuthAppConfig.RedirectURIs",
+			},
+		},
+	}
```

Comment:
this can be moved to different go files as template_rule.go

Reply:
This was injected to parameterizer to support
1. If the rules needs to be configurable in future. 
2. Test the parameterizer with different scenarios.